### ISSUE TITLE
Salesforce support

### ISSUE THREAD

**Issue by dannyeuu** [metabase/metabase#8366]
### Salesforce feature request
It will be a great feature to use Salesforce data like actually, we use Google Analytics data.

**Comment by LudoTheHUN** [comment]
No need https://github.com/metabase/metabase/issues/8701 :-O

**Comment by chucksense** [comment]
It would still be useful to be able to pull data from SalesForce such that SalesForce data can be embedded side-by-side in dashboards with other data sources.

**Comment by emeth-** [comment]
Agreed with this - Salesforce even supports SOQL (a SQL-like language) over their API, which returns json objects. 

You can see there is significant interest in the project just from the number of clicks on the link in this thread:
https://discourse.metabase.com/t/easiest-way-to-get-salesforce-data-into-metabase/462/2

I started to take a stab at implementing this myself, basing it off of https://github.com/tlrobinson/metabase-http-driver but unfortunately was unable to get that repository to compile due to java.xml.bind issues. 

I see that the main metabase repo injects javax.xml.bind via dependencies, but this metabase-http-driver injects it via a jvm-opts of --add-modules=java.xml.bind. I attempted to adjust it to match the main metabase repo, but still could not get past the error.

If anybody knows how to update the metabase-http-driver repo to resolve the java.xml.bind issue, I'd be happy to pick this up again and take another stab once that issue is resolved. (Fyi @tlrobinson if you're still around)

**Comment by flamber** [comment]
@emeth- Many drivers needs a few changes to work in 0.34 and again in 0.35. You might want to have look at the documentation and some of the referenced drivers that are updated.
https://www.metabase.com/docs/latest/developers-guide-drivers.html

**Comment by emeth-** [comment]
Thanks @flamber, I'll check that out. 

Reviewing the link you posted, I see this:
> If your database has a JDBC driver, you’ll be able to make use of some common classes that already exist in Metabase.

Apparently there is an MIT-licensed JDBC driver for Salesforce already, so that could be a better path than basing it off the JSON driver:
https://github.com/ascendix/salesforce-jdbc

So I'll try going down that road instead and basing it off a more recently updated driver that also uses JDBC. Thanks for the tip.
